presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -592,6 +593,17 @@ public static NullableValue parsePartitionValue(HivePartitionKey key, Type type,
         return parsePartitionValue(key.getName(), key.getValue().orElse(HIVE_DEFAULT_DYNAMIC_PARTITION), type, timeZone);
     }
 
+    public static NullableValue parsePartitionValue(String partitionName, String value, Type type, ZoneId hiveStorageTimeZoneId)
```

Comment:
Just thinking out loud, NullableValue is deprecated. Can we avoid using it with these changes? @harris233

Reply:
This implementation refers to hive connector. It has the same problem in hive.